### PR TITLE
Introduce x-aws-cluster for cluster to deploy application to

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
Introduced `x-aws-cluster` to let user define the cluster to deploy to as a compose extension

**Related issue**
https://github.com/docker/ecs-plugin/issues/169

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**
image

### CODE REVIEW

**Comment by ndeloof** [comment]
Known limitation doing so: `ps` command require the cluster to invoke `DescribeServices` API, so we can't run `ps` by just application name, need the original compose file

File: pkg/amazon/backend/context.go [hunk 1]
```diff
@@ -11,14 +12,8 @@ func (b *Backend) CreateContextData(ctx context.Context, params map[string]strin
...
-			return "", "", err
-		}
-		if !exists {
-			return "", "", fmt.Errorf("cluster %s does not exists", b.Cluster)
-		}
-	}
-	return "", "", nil
+	return docker.AwsContext{
+		Profile: params["profile"],
+		Region:  params["regsion"],
```
Comment (chris-crone) [inline pkg/amazon/backend/context.go:17]:
```suggestion
		Region:  params["region"],
```

File: pkg/amazon/backend/up.go [hunk 2]
```diff
@@ -16,14 +16,14 @@ func (b *Backend) Up(ctx context.Context, options cli.ProjectOptions) error {
...
-		if err != nil {
-			return err
-		}
-		if !ok {
-			return fmt.Errorf("configured cluster %q does not exist", b.Cluster)
-		}
+	err = b.api.CheckRequirements(ctx)
+	if err != nil {
+		return err
+	}
```
Comment (chris-crone) [inline pkg/amazon/backend/up.go:22]:
```suggestion
	if err := b.api.CheckRequirements(ctx); err != nil {
		return err
	}
```

File: pkg/amazon/backend/down.go [hunk 3]
```diff
@@ -9,13 +9,9 @@ import (
...
 
 func (b *Backend) Down(ctx context.Context, options cli.ProjectOptions) error {
-	name := options.Name
-	if name == "" {
-		project, err := cli.ProjectFromOptions(&options)
-		if err != nil {
-			return err
-		}
-		name = project.Name
+	name, err2 := b.projectName(options)
```
Comment (chris-crone) [inline pkg/amazon/backend/down.go:12]: Why can't we use `err` here?

Comment (ndeloof) [reply pkg/amazon/backend/down.go:12]: IDE refactoring

File: pkg/amazon/backend/up.go (hunk 2 above)
Comment (ndeloof) [reply pkg/amazon/backend/up.go:22]: I hate this pattern. It doesn't make things more readable, wrapping the actual func call into boilerplate error management code, and make it harde to refactor if we later want to change func signature to return a value in addition to `err`. Also, my IDE do collapse all the err management shit as a single line, but won't if I adopt this.

**Comment by ndeloof** [comment]
f516af8 is actually related: DescribeService API require cluster name as parameter.

**Pull request merged**